datasets/us/ofac/enforcement_actions/crawler.py

```diff
@@ -0,0 +1,291 @@
+from datetime import date
+from lxml.html import HtmlElement, fromstring, tostring
+from pydantic import BaseModel, Field
+from rigour.mime.types import HTML
+from typing import Optional, List, Literal
+
+from zavod import Context
+from zavod import helpers as h
+from zavod.entity import Entity
+from zavod.shed import enforcements
+from zavod.shed.gpt import DEFAULT_MODEL, run_typed_text_prompt
+from zavod.stateful.review import (
+    Review,
+    assert_all_accepted,
+    request_review,
+    get_review,
+    model_hash,
+    html_to_text_hash,
+)
+
+NAME_XPATH = "//span[@class='treas-page-title']/text()"
+CONTENT_XPATH = "//div[@id='block-ofac-content']//div[@class='content']"
+DATE_XPATH = "//div[@id='block-ofac-content']//div[@class='field__item']/text()"
+# Traveling back to summer of 2016 (the first full pdf enforcement was issued on July 5, 2016)
+MAX_AGE_DAYS = (date.today() - date(2016, 7, 8)).days
+
+Schema = Literal["Person", "Company", "LegalEntity"]
+Status = Literal[
+    "Filed",
+    "Dismissed",
+    "Settled",
+    "Final judgement",
+    "Other",
+]
+
+# Settlement Agreement
+# Penalty Notice to an Individual
+# Finding of Violation
+# Sanctions Compliance Guidance
+# Issuance of Sanctions Regulations
+# Civil Monetary Penalty
+# Release of OFAC Enforcement Information
+# Release​ of Civil Penalties Information
+# Recent OFAC Actions
+
+MODEL_VERSION = 1
+MIN_MODEL_VERSION = 1
+
+something_changed = False
+
+schema_field = Field(
+    description="Use LegalEntity if it isn't clear whether the entity is a person or a company."
+)
+address_field = Field(
+    default=[],
+    description=("The addresses or even just the districts/states of the defendant."),
+)
+status_field = Field(
+    description=(
+        "The status of the enforcement action notice."
+        " If `Other`, add the text used as the status in the source to `notes`."
+    )
+)
+notes_field = Field(default=None, description=("Only used if `status` is `Other`."))
+
+
+class RelatedCompany(BaseModel):
+    name: str
+    relationship: str
+
+
+class Defendant(BaseModel):
+    entity_schema: Schema = schema_field
+    name: str
+    aliases: List[str] = []
+    address: str | List[str] | None = address_field
+    country: str | List[str] | None = []
+    status: Status = status_field
+    notes: Optional[str] = notes_field
+    related_companies: List[RelatedCompany] = []
+    pdf_url: Optional[str] = None
```

Comment:
nice - could make it a list defaulting to [] from the start (I'm finding more and more things that want to be a list in my crawlers). Then I imagine we can expect to see both links on https://ofac.treasury.gov/recent-actions/20241203_33 added, and just need to check that the result for unicredit is still good

Reply:
Just remember to update the prompt too